### PR TITLE
Bunch of gas optimizations

### PR DESCRIPTION
**Pull request by akshatmittal**
1. Resolves https://github.com/code-423n4/2023-06-reserve-findings/issues/27
2. Resolves https://github.com/code-423n4/2023-06-reserve-findings/issues/50
3. Optimizes Events

### CODE REVIEW

**Review by pmckelvy1 — APPROVED** [review]
lgtm

**Comment by pmckelvy1** [comment]
can you post a before/after gas diff?

File: contracts/p1/BasketHandler.sol [hunk 1]
```diff
@@ -255,7 +255,12 @@ contract BasketHandlerP1 is ComponentP1, IBasketHandler {
 
         for (uint256 i = 0; i < size; ++i) {
             CollateralStatus s = assetRegistry.toColl(basket.erc20s[i]).status();
-            if (s.worseThan(status_)) status_ = s;
+            if (s.worseThan(status_)) {
+                status_ = s;
+            }
+            if (status_ == CollateralStatus.DISABLED) {
```
Comment (tbrent) [inline contracts/p1/BasketHandler.sol:261]: Not sure you saw my comment in the other PR but I think this change will end up being a net gas increase given how infrequently one of the backing will be DISABLED.

A change I like that might be the best of both worlds is putting this check inside the if statement above

Comment (akshatmittal) [reply contracts/p1/BasketHandler.sol]: You are right, I was going to remove this, just missed it.

**Comment by tbrent** [comment]
As expected, the gas improvements are extremely marginal

**Pull request merged**